### PR TITLE
Picker API review

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Update Pickers API.

#### WHY

Follow idiomatic compose conventions.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

#### Later
- [ ] Confirm a11y for AM/PM
- [ ] Add chevron and next for date picker 
- [ ] CompoundPicker and State for a reusable Picker base.

### CODE REVIEW

File: composables/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -22,4 +22,5 @@
     <string name="picker_day">Day</string>
     <string name="picker_month">Month</string>
     <string name="picker_year">Year</string>
+    <string name="picker_check_button">Check</string>
```
Comment (yschimke) [inline composables/src/main/res/values/strings.xml:25]: Is "Check" right? Confirm?

File: composables/src/main/java/com/google/android/horologist/composables/DatePicker.kt [hunk 2]
```diff
@@ -201,10 +202,16 @@ public fun DatePicker(
...
                             dayState.selectedOption + 1
                         )
-                        onClick(date)
+                        onValueConfirm(date)
                     },
                 ) {
-                    buttonIcon()
+                    Icon(
+                        imageVector = Icons.Filled.Check,
+                        contentDescription = stringResource(id = R.string.picker_check_button),
```
Comment (ssancho14) [inline composables/src/main/java/com/google/android/horologist/composables/DatePicker.kt:210]: The strings namespace is shared between the app and all libraries, right? May make sense to prefix with the library name?

File: composables/src/main/res/values/strings.xml (hunk 1 above)
Comment (ssancho14) [reply composables/src/main/res/values/strings.xml:25]: Not sure. I think I prefer Confirm, to match `onValueConfirm`. Other alternatives: Accept, Ok

File: composables/src/main/java/com/google/android/horologist/composables/TimePicker.kt [hunk 3]
```diff
@@ -73,18 +76,16 @@ import java.time.temporal.ChronoField
  * overrides for MaterialTheme.typography.display3 which is used to display the main picker
  * value.
  *
- * @param buttonIcon the button content.
- * @param onClick the button event handler.
+ * @param onValueConfirm the button event handler.
  * @param modifier the modifiers for the `Box` containing the UI elements.
- * @param initial the initial value to seed the picker with.
+ * @param value the initial value to seed the picker with.
```
Comment (ssancho14) [inline composables/src/main/java/com/google/android/horologist/composables/TimePicker.kt:81]: Ok, rereading https://developer.android.com/jetpack/compose/state#state-hoisting , seems that you can use a name that makes sense for the type of data, so if you want to rename this parameter `time`, and the function `onTimeChanges`/`onTimeConfirmed` that will be fine too.
On the other hand, if all these pickers have different names for the parameters may get confusing.

File: composables/src/main/java/com/google/android/horologist/composables/DatePicker.kt [hunk 4]
```diff
@@ -60,18 +63,16 @@ import java.time.temporal.TemporalAdjusters
...
  */
 @ExperimentalHorologistComposablesApi
 @Composable
 public fun DatePicker(
-    buttonIcon: @Composable BoxScope.() -> Unit,
-    onClick: (LocalDate) -> Unit,
+    onValueConfirm: (LocalDate) -> Unit,
     modifier: Modifier = Modifier,
-    initial: LocalDate = LocalDate.now()
+    value: LocalDate = LocalDate.now()
```
[the remark this answers is not in the document]
Comment (yschimke) [reply composables/src/main/java/com/google/android/horologist/composables/DatePicker.kt:75]: Yep, conscious decision. Plus apps should be able to use https://developer.android.com/studio/write/java8-support-table

File: composables/src/main/java/com/google/android/horologist/composables/DatePicker.kt (hunk 2 above)
Comment (yschimke) [reply composables/src/main/java/com/google/android/horologist/composables/DatePicker.kt:210]: Makes sense, I'll follow up in a PR and do all together.

File: composables/src/main/java/com/google/android/horologist/composables/TimePicker.kt (hunk 3 above)
Comment (yschimke) [reply composables/src/main/java/com/google/android/horologist/composables/TimePicker.kt:81]: Let's go woth date and time then

**Review by luizgrp — APPROVED** [review]

**Pull request merged**